src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -359,50 +376,170 @@ internal async Task<IEnumerable<JObject>> GetFunctionTriggers(IEnumerable<Functi
+
+            var durableExtension = extensionItems.FirstOrDefault(ext => string.Equals(ext.Name, "DurableTask", StringComparison.OrdinalIgnoreCase));
+            if (durableExtension == null)
+            {
+                return null;
+            }
+
+            var versionMatch = versionRegex.Match(durableExtension.TypeName);
+            if (!versionMatch.Success)
+            {
+                return null;
+            }
+
+            // Grab the captured group.
+            return versionMatch.Groups["majorversion"].Captures[0].Value;
+        }
+
+        private GlobalDurableConfig GetDurableV1Config(JObject durableHostConfig)
+        {
+            var config = new GlobalDurableConfig();
+            try
+            {
+                if (durableHostConfig.TryGetValue(HubName, StringComparison.OrdinalIgnoreCase, out JToken nameValue) && nameValue != null)
+                {
+                    config.HubName = nameValue.ToString();
+                }
+
+                if (durableHostConfig.TryGetValue(DurableTaskV1StorageConnectionName, StringComparison.OrdinalIgnoreCase, out nameValue) && nameValue != null)
+                {
+                    config.Connection = nameValue.ToString();
+                }
+            }
+            catch (Exception)
+            {
+                throw new InvalidDataException(InvalidDurableV1ConfigErrorMsg);
+            }
+
+            return config;
+        }
+
+        private GlobalDurableConfig GetDurableV2Config(JObject durableHostConfig)
+        {
+            var config = new GlobalDurableConfig();
+            try
+            {
+                if (durableHostConfig.TryGetValue(HubName, StringComparison.OrdinalIgnoreCase, out JToken nameValue) && nameValue != null)
+                {
+                    config.HubName = nameValue.ToString();
+                }
+                else
+                {
+                    string hubName = GetDefaultDurableV2HubName();
+                    config.HubName = hubName;
+                }
+
+                if (durableHostConfig.TryGetValue(DurableTaskV2StorageOptions, StringComparison.OrdinalIgnoreCase, out JToken storageOptions) && (storageOptions as JObject) != null)
+                {
+                    if (((JObject)storageOptions).TryGetValue(DurableTaskV2StorageConnectionName, StringComparison.OrdinalIgnoreCase, out nameValue) && nameValue != null)
                     {
-                        throw new InvalidDataException("Invalid host.json configuration for 'durableTask'.");
+                        config.Connection = nameValue.ToString();
                     }
                 }
             }
+            catch (Exception)
+            {
+                throw new InvalidDataException(InvalidDurableV2ConfigErrorMsg);
+            }
 
             return config;
         }
 
+        // This logic will eventually be moved to ScaleController once it has access to version information.
```

Comment:
And for the runtime driven scale work [here](https://github.com/Azure/azure-functions-durable-extension/commit/384ee5b68b8ef77747d4e7c770bbc39ae8c0a5df) it'll already do the right thing?

Reply:
Our goal should be to remove all of this Durable specific logic from SyncTriggers so it behaves like all the other extensions, and the host doesn't have to have any specific knowledge of Durable. Perhaps that can be done by having Durable write some hint in the storage account, or otherwise have the ScaleController sniff the storage account entities to determine what to do?

Reply:
Runtime scale is doing the right thing, as it is grabbing information directly from the extension.

As far as your other idea, I think that is definitely something that could work in the long term. @cgillum, thoughts?

Reply:
I logged https://github.com/Azure/azure-functions-host/issues/5390 to track this